UnityProject/Storybook/Assets/Code/Scripts/Game/Player/PlayerMover.cs

```diff
@@ -0,0 +1,338 @@
+            }
+
+            // Only allow the player to select a door if the door is enabled for the current room 
+            if (isSouthDoorEnabled && GUILayout.Button("Move Down"))
+            {
+                int newRoomLocX = m_currentRoomLoc.X + 1;
+                int newRoomLocY = m_currentRoomLoc.Y;
+                photonView.RPC("CreateRoomOnMaster", PhotonTargets.MasterClient, newRoomLocX, newRoomLocY, RoomObject.DoorIndex.SouthDoor);
+            }
+        }
+    }
+
+    /// <summary>
+    /// The list of players that are currently in the dungeon
+    /// </summary>
+    public List<NetworkNodeMover> WorldPlayers
+    {
+        get { return m_worldPlayers; }
+        set { m_worldPlayers = value; }
+    }
+
+    /// <summary>
+    /// Sets the target nodes of the all the players to the corresponding player nodes
+    /// </summary>
+    private void _SetTargetNodesForPlayers()
+    {
+        if (m_worldPlayers.Count >= 1)
+        {
+            m_worldPlayers[0].TargetNode = m_currentRoom.Player1Node;
+            m_worldPlayers[0].IsAtTarget = false;
+        }
+        if (m_worldPlayers.Count >= 2)
+        {
+            m_worldPlayers[1].TargetNode = m_currentRoom.Player2Node;
+            m_worldPlayers[1].IsAtTarget = false;
+        }
+        if (m_worldPlayers.Count >= 3)
+        {
+            m_worldPlayers[2].TargetNode = m_currentRoom.Player3Node;
+            m_worldPlayers[2].IsAtTarget = false;
+        }
+        if (m_worldPlayers.Count >= 4)
+        {
+            m_worldPlayers[3].TargetNode = m_currentRoom.Player4Node;
+            m_worldPlayers[3].IsAtTarget = false;
+        }
+    }
+
+    /// <summary>
+    /// Moves the camera to the new position and rotation
+    /// Called when the players switch rooms
+    /// </summary>
+    /// <param name="newCamPos">The new position for the camera</param>
+    /// <param name="newCamRot">The new rotation for the camera</param>
+    [PunRPC]
+    public void SwitchCameraRoom(Vector3 newCamPos, Quaternion newCamRot)
+    {
+        Camera.main.transform.position = newCamPos;
+        Camera.main.transform.rotation = newCamRot;
+    }
+
+    /// <summary>
+    /// Sends the current room data to the clients
+    /// </summary>
+    /// <param name="isNorthDoorActive">True if the north door is active for the current room, false otherwise</param>
+    /// <param name="isEastDoorActive">True if the east door is active for the current room, false otherwise</param>
+    /// <param name="isSouthDoorActive">True if the south door is active for the current room, false otherwise</param>
+    /// <param name="isWestDoorActive">True if the west door is active for the current room, false otherwise</param>
+    /// <param name="roomType">The type of the room (MapManager.RoomType)</param>
+    [PunRPC]
+    public void SendRoomData(bool isNorthDoorActive, bool isEastDoorActive, bool isSouthDoorActive, bool isWestDoorActive, int roomType)
```

Comment:
This is not the class to handle sending room data. Room objects can be made as network objects. Then just give them SyncProperties for the door properties. Its easy and flexible.